Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/CorrelationAttributeUtil.java

```diff
@@ -137,54 +139,56 @@ private static String getEmailAddressAttrDisplayName() {
-                } else if (parent instanceof OsAccount) {
-                    for (OsAccountInstance osAccountInst : ((OsAccount) parent).getOsAccountInstances()) {
-                        if (osAccountInst.getDataSource().equals(analysisResult.getDataSource())) {
-                            correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch(osAccountInst));
-                            break;
+                } else {
+                    if (artifactTypeID == ARTIFACT_TYPE.TSK_KEYWORD_HIT.getTypeID()) {
+                        BlackboardAttribute setNameAttr = analysisResult.getAttribute(BlackboardAttribute.Type.TSK_SET_NAME);
+                        if (setNameAttr != null && CorrelationAttributeUtil.getEmailAddressAttrDisplayName().equals(setNameAttr.getValueString())) {
+                            correlationAttrs.addAll(makeCorrAttrFromArtifactAttr(analysisResult, BlackboardAttribute.ATTRIBUTE_TYPE.TSK_KEYWORD, CorrelationAttributeInstance.EMAIL_TYPE_ID, analysisResult.getAttributes()));
+                        }
+                    }
+                    Content parent = analysisResult.getParent();
+                    if (parent instanceof AbstractFile) {
+                        correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch((AbstractFile) parent));
+                    } else if (parent instanceof AnalysisResult) {
+                        correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch((AnalysisResult) parent));
+                    } else if (parent instanceof DataArtifact) {
+                        correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch((DataArtifact) parent));
+                    } else if (parent instanceof OsAccount) {
+                        for (OsAccountInstance osAccountInst : ((OsAccount) parent).getOsAccountInstances()) {
+                            if (osAccountInst.getDataSource().equals(analysisResult.getDataSource())) {
+                                correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch(osAccountInst));
+                                break;
+                            }
                         }
                     }
                 }
-            }
 
-        } catch (TskCoreException ex) {
-            logger.log(Level.SEVERE, "Failed to get information regarding correlation attributes from AnalysisResult", ex);
-        } catch (NoCurrentCaseException ex) {
-            logger.log(Level.SEVERE, "Attempted to retrieve correlation attributes for search with no currently open case.", ex);
-        } catch (CentralRepoException ex) {
-            logger.log(Level.SEVERE, "Failed to get correlation type from central repository.", ex);
+            } catch (TskCoreException ex) {
+                logger.log(Level.SEVERE, "Failed to get information regarding correlation attributes from AnalysisResult", ex);
```

Comment:
Let's make sure this message is accurate, i.e., that the exception can only be thrown by the analysis result case.

Reply:
It was somewhat accurate since this whole method is for the AnaylsisResult, but I have adjusted the language used to clarify that it could be the provided AnalysisResult, its associated artifact, or its parent's use that led to this exception.